### PR TITLE
Update artifact location filter with tally

### LINKED ISSUE #898: Artifact Location/Equipped/Inventory tally
image

New tally system is cool. But should these buttons also include a tally?

### LINKED ISSUE #899: Artifact Equipped/Inventory logic
Currently the UI logic seems a little strange for the "equipped" button as it seems to be acting more like a toggle to enable/disable the "location" multi-select drop down.

It's a little counter intuitive since you have to turn "off" showing equipped to then be able to define which characters to show equipped for. There might be a better way to communicate that in the UI?

The terminology "Inventory" can also be confusing as artifacts in your in-game inventory do not discriminate between equipped status. It's just all the artifacts in your inventory.

States you want is:
All Artifacts (Equipped and Unequipped)
Exclusively Equipped
Exclusively Unequipped
Exclusively Character Specific (Multi-select)
Character Specific and Unequipped

How about the following:
image

I think this communicates the intention clearer and the UI flows a little more when "toggling" the All Equipped button to enable the Location field below

### PR DESCRIPTION
**Pull request by frzyc**
Resolve #898
Resolve #899
image
image

### CODE REVIEW

File: apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx [hunk 1]
```diff
@@ -95,6 +115,14 @@ export default function ArtifactFilterDisplay({ filterOption, filterOptionDispat
     })
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx:115]: Pre-existing, but could you fix the warnings for `artSubTotal`?
image

File: apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx [hunk 2]
```diff
@@ -141,9 +171,14 @@ export default function ArtifactFilterDisplay({ filterOption, filterOptionDispat
...
       <ArtifactMainStatMultiAutocomplete totals={artMainTotal} mainStatKeys={mainStatKeys} setMainStatKeys={mainStatKeys => filterOptionDispatch({ mainStatKeys })} />
       <ArtifactSubstatMultiAutocomplete totals={artSubTotal} substatKeys={substats} setSubstatKeys={substats => filterOptionDispatch({ substats })} />
-      <LocationFilterMultiAutocomplete locations={showEquipped ? [] : locations} setLocations={locations => filterOptionDispatch({ locations })} disabled={showEquipped} />
-      <Button startIcon={<PersonSearch />} color={showEquipped ? "success" : "secondary"} onClick={() => filterOptionDispatch({ showEquipped: !showEquipped })}>{t`equippedArt`}</Button>
-      <Button startIcon={<BusinessCenter />} color={showInventory ? "success" : "secondary"} onClick={() => filterOptionDispatch({ showInventory: !showInventory })}>{t`artInInv`}</Button>
+      <Suspense fallback={null}>
+        <Tooltip title={t`locationsTooltip`} placement="top" open={showEquipped ? locationsTooltipOpen : false} disableInteractive arrow>
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx:175]: Use `BootstrapTooltip`

File: apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx [hunk 3]
```diff
@@ -141,9 +171,14 @@ export default function ArtifactFilterDisplay({ filterOption, filterOptionDispat
...
       <ArtifactMainStatMultiAutocomplete totals={artMainTotal} mainStatKeys={mainStatKeys} setMainStatKeys={mainStatKeys => filterOptionDispatch({ mainStatKeys })} />
       <ArtifactSubstatMultiAutocomplete totals={artSubTotal} substatKeys={substats} setSubstatKeys={substats => filterOptionDispatch({ substats })} />
-      <LocationFilterMultiAutocomplete locations={showEquipped ? [] : locations} setLocations={locations => filterOptionDispatch({ locations })} disabled={showEquipped} />
-      <Button startIcon={<PersonSearch />} color={showEquipped ? "success" : "secondary"} onClick={() => filterOptionDispatch({ showEquipped: !showEquipped })}>{t`equippedArt`}</Button>
-      <Button startIcon={<BusinessCenter />} color={showInventory ? "success" : "secondary"} onClick={() => filterOptionDispatch({ showInventory: !showInventory })}>{t`artInInv`}</Button>
+      <Suspense fallback={null}>
+        <Tooltip title={t`locationsTooltip`} placement="top" open={showEquipped ? locationsTooltipOpen : false} disableInteractive arrow>
+          <Box onMouseOver={() => setlocationsTooltipOpen(true)} onMouseLeave={() => setlocationsTooltipOpen(false)} >
```
Comment (nguyentvan7) [inline apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx:176]: You don't need to do all this `Box` with state funkiness. Wrapping the autocomplete in a span is enough, and preserves mobile tooltip functionality:

```
        <BootstrapTooltip title={showEquipped? t`locationsTooltip` : ""} placement="top">
          <span>
            <LocationFilterMultiAutocomplete totals={locationTotal} locations={showEquipped ? [] : locations} setLocations={locations =>
              filterOptionDispatch({ locations })} disabled={showEquipped} />
          </span>
        </BootstrapTooltip>
```

**Review by nguyentvan7 — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx (hunk 3 above)
Comment (frzyc) [reply apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx:176]: The intention here is to only show the tooltip when the location autocomplete is disabled.

Comment (nguyentvan7) [reply apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx]: My code does that. 

`title={showEquipped ? t("locationsTooltip") : ""}`

a blank `title` disables the tooltip automatically

Comment (frzyc) [reply apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx]: huh it didnt work for me, I guess it didnt hotload. I'll make the change, good catch.

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**